plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/ReadSessionCreator.java

```diff
@@ -130,7 +122,7 @@ private TableInfo getActualTable(
             String query = bigQueryClient.selectSql(table.getTableId(), requiredColumns);
             log.debug("query is %s", query);
             try {
-                return destinationTableCache.get(query, new DestinationTableBuilder(bigQueryClient, config, query, table.getTableId()));
+                return bigQueryClient.getDestinationTableCache().get(query, new DestinationTableBuilder(bigQueryClient, config, query, table.getTableId()));
```

Comment:
I think we could move DestinationTableBuilder to BigQueryClient and expose a method `bigQueryClient.getCachedTable(config, table.getTableId(), requiredColumns)` instead of exposing the cache directly.

Reply:
Do you see any issue in exposing the cache?

Reply:
Done.